packages/execution-environments/src/common/commands.test.ts

```diff
@@ -0,0 +1,168 @@
+      onTerminate,
+    );
+    expect(await methodsObj.ping()).toStrictEqual('OK');
+  });
+
+  it("the terminate method will 'OK'", async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    expect(await methodsObj.terminate()).toStrictEqual('OK');
+  });
+
+  it("the executeSnap method will return 'OK'", async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    expect(
+      await methodsObj.executeSnap('foo', 'code', ['endowment1', 'endowment2']),
+    ).toStrictEqual('OK');
+  });
+
+  it('the executeSnap method will throw an Error if the snapName is not a string', async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    await expect(async () => {
+      await methodsObj.executeSnap(1 as any, 'code', [
+        'endowment1',
+        'endowment2',
+      ]);
+    }).rejects.toThrow('snapName is not a string');
+  });
+
+  it('the executeSnap method will throw an Error if the sourceCode is not a string', async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    await expect(async () => {
+      await methodsObj.executeSnap('foo', 2 as any, [
+        'endowment1',
+        'endowment2',
+      ]);
+    }).rejects.toThrow('sourceCode is not a string');
+  });
+
+  it('the executeSnap method will throw an Error if it is not passed a proper Endowments object', async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    await expect(async () => {
+      await methodsObj.executeSnap('foo', 'code', ['endowment1', 2 as any]);
+    }).rejects.toThrow('endowment is not a proper Endowments object');
+  });
+
+  it('the snapRpc method will invoke the invokeSnapRpc function', async () => {
+    const startSnap = jest.fn();
+    const invokeSnapRpc = jest.fn();
+    const onTerminate = jest.fn();
+    const methodsObj = getCommandMethodImplementations(
+      startSnap,
+      invokeSnapRpc,
+      onTerminate,
+    );
+    const rpcRequest = { jsonrpc: '2.0', method: 'hello' };
+    await methodsObj.snapRpc('foo', 'bar', rpcRequest as any);
+    expect(invokeSnapRpc).toHaveBeenCalledTimes(1);
```

Comment:
Could we add `toHaveBeenCalledWith` and verify the params?

Reply:
yup!